src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -706,3 +714,56 @@ def create_python_udf_or_sp(
     telemetry_client.send_function_usage_telemetry(
         api_call_source, TelemetryField.FUNC_CAT_CREATE.value
     )
+
+
+def generate_anonymous_python_sp_sql(
```

Comment:
This probably shares a lot with `create_python_udf_or_sp` for sql generation, I wonder if we could refactor a bit to share more code?

Reply:
I also thought it before, but actually they are not really the same (e.g., object_type, execute_as, and many other arguments cannot apply to anonymous stored proc), so I prefer to keep them separate first. Once we want to add public support, we can check it later to see whether there is a refactoring opportunity.

Reply:
Sure, https://snowflakecomputing.atlassian.net/browse/SNOW-769761